### DISCUSSION TITLE
[Feature Request] semantic-dom-diff inline snapshots

### DISCUSSION THREAD

**Discussion by michaelwarren1106** [open-wc/open-wc#2323] [Ideas]
Currently the [Semantic Dom Diff](https://github.com/open-wc/open-wc/tree/master/packages/semantic-dom-diff) package has a `chai` assertion for `equalsSnapshot()` where the snapshots get stored in folders/files.

In jest land, I've used the `toMatchInlineSnapshot` feature extensively so as to not have to push/maintain snapshot files in the repo as a separate source code asset.

I'd like to propose that the inline snapshot feature be added to `semantic-dom-diff` so that tests can just inline the snapshots in code and not need to have external snapshot files in the name of ease of use/less files, and also in the name of jest parity.

**Comment by LarsDenBakker** [comment]
This should already be possible using .equal() on an expect/asset using .shadowDom etc.

**Reply by michaelwarren1106** [reply]
ok cool! it didnt really seem like the same concept as "snapshots" so i didnt connect those as being the same as "inline" snapshots... maybe in the name of jest parity maybe the documentation could be organized to call out explicitly that "inline" snapshots" have an approach and give code examples?

**Reply by michaelwarren1106** [reply]
actually, would you mind pasting a quick code example?

looking through the docs, i see this kind of example:
```expect(el).shadowDom.to.equal('<div></div>');```

but this is not really like a snapshot, since the `<div></div>` string isnt generated ahead of time, so with this example you couldnt test that your shadowDom hadnt changed from the last pull request like the file-based snapshots allow for.

imo, in order for inline snapshots to be possible, the dom string you're comparing to needs to be saved off ahead of time so that current changes can be judged against that static version. Also there would need to be some mechanism to update the inline snapshot.

in jest, the `toMatchInlineSnapshot` feature actually changes the source code of the files before the tests run and updates that static snapshot comparison if the `-u` flag is passed in the test script command. IMO, to get inline snapshots working there would need to be other custom build scripts created so that `.shadowDom.to.equal(someString)` would work.
